src/core/src/service_interfaces/TelemetryWriter.py

```diff
@@ -273,8 +273,8 @@ def __get_events_dir_size(self):
         """ Returns total size, in bytes, of the events folder """
         total_dir_size = 0
         for f in os.listdir(self.events_folder_path):
-            if os.path.isfile(os.path.join(self.events_folder_path, f)):
-                total_dir_size += os.path.getsize(os.path.join(self.events_folder_path, f))
+            file_path = os.path.join(self.events_folder_path, f)
+            total_dir_size += os.path.getsize(file_path) if os.path.exists(file_path) and os.path.isfile(file_path) else 0
```

Comment:
Is the exists check necessary? https://docs.python.org/3/library/os.path.html#os.path.isfile may do it implicitly if I am reading the documentation correctly.

Reply:
More investigation required, because the error re-appeared while trying this suggestion

Reply:
Neither solution (first fix or comment on removing the exists check) guarantees the underlying issue will go away. 

You'll always be subject to timing issues in a path check + operation (as it's not atomic) and the thing that guarantees safety to do is to wrap getsize in a try catch - log further if the exception is not related to the file missing; and continue. Don't add anything to the dir size if it throws.

Reply:
Resolved